### PR TITLE
layers: Reduce looping shader for decoration

### PR DESCRIPTION
**Pull request by sfricke-samsung**
There are many cases where we are looping the entire module just to fine the `OpMemberDecorate` and `OpDecorate`

### CODE REVIEW

**Comment by ncesario-lunarg** [comment]
Related to https://github.com/KhronosGroup/Vulkan-ValidationLayers/projects/4#card-50486050.

File: layers/shader_validation.h [hunk 1]
```diff
@@ -191,6 +191,9 @@ struct SHADER_MODULE_STATE : public BASE_NODE {
     // trees, constant expressions, etc requires jumping all over the instruction stream.
     std::unordered_map<unsigned, unsigned> def_index;
     std::unordered_map<unsigned, decoration_set> decorations;
+    // Find all decoration instructions without needing to reloop modle as many checks need this info
```
Comment (sfricke-samsung) [inline layers/shader_validation.h:194]: typo :facepalm:

**Comment by ncesario-lunarg** [comment]
Running through [internal CI](http://erusea:4040/8432).

**Review by ncesario-lunarg — APPROVED** [review]
Looks great!

**Pull request merged**